hbase-server/src/main/java/org/apache/hadoop/hbase/mob/RSMobFileCleanerChore.java

```diff
@@ -0,0 +1,262 @@
+              //collectinng referenced MOBs
+              for (HStoreFile sf : sfs) {
+                currentPath = sf.getPath();
+                sf.initReader();
+                byte[] mobRefData = sf.getMetadataValue(HStoreFile.MOB_FILE_REFS);
+                byte[] bulkloadMarkerData = sf.getMetadataValue(HStoreFile.BULKLOAD_TASK_KEY);
+                // close store file to avoid memory leaks
+                sf.closeStoreFile(true);
+                if (mobRefData == null) {
+                  if (bulkloadMarkerData == null) {
+                    LOG.warn("Found old store file with no MOB_FILE_REFS: {} - "
+                      + "can not proceed until all old files will be MOB-compacted.", currentPath);
+                    return;
+                  } else {
+                    LOG.debug("Skipping file without MOB references (bulkloaded file):{}",
+                      currentPath);
+                    continue;
+                  }
+                }
+                // file may or may not have MOB references, but was created by the distributed
+                // mob compaction code.
+                try {
+                  SetMultimap<TableName, String> mobs =
+                    MobUtils.deserializeMobFileRefs(mobRefData).build();
+                  LOG.debug("Found {} mob references for store={}", mobs.size(), sf);
+                  LOG.trace("Specific mob references found for store={} : {}", sf, mobs);
+                  regionMobs.addAll(mobs.values());
+                } catch (RuntimeException exception) {
+                  throw new IOException("failure getting mob references for hfile " + sf,
+                    exception);
+                }
+              }
+              //collecting files, MOB included currently being written
+              regionMobs.addAll(
+                store.getStoreFilesBeingWritten().stream().map(path -> path.getName())
+                  .collect(Collectors.toList()));
+
+              referencedMOBs.computeIfAbsent(hcd.getNameAsString(),
+                  cf -> new HashMap<String, List<String>>())
+                .computeIfAbsent(region.getRegionInfo().getEncodedName(), name -> new ArrayList<>())
+                .addAll(regionMobs);
+
+            } catch (FileNotFoundException e) {
+              LOG.warn(
+                "Missing file:{} Starting MOB cleaning cycle from the beginning" + " due to error",
+                currentPath, e);
+              regionMobs.clear();
+              continue;
+            } catch (IOException e) {
+              LOG.error("Failed to clean the obsolete mob files for table={}",
+                htd.getTableName().getNameAsString(), e);
+            }
+          }
+        }
+
+        LOG.debug("Found: {} active mob refs for table={}",
+          referencedMOBs.values().stream().map(inner -> inner.values())
+            .flatMap(lists -> lists.stream()).mapToInt(lists -> lists.size()).sum(),
+          htd.getTableName().getNameAsString());
```

Comment:
nit: check if debug is enabled, otherwise we'll always be computing the debug param.

Reply:
That is a very nice catch, thanks!